internal/monitors/device-updater/device_updater.go

```diff
@@ -47,42 +50,119 @@ func (d *DeviceUpdater) UpdateDevices() {
 		return
 	}
 
-	for _, fleet := range fleets {
+	for fleetIndex := range fleets {
+		orgID := fleets[fleetIndex].OrgID
 		// If there is no template set in the fleet, then there is nothing to sync to the devices
-		if fleet.Spec == nil || fleet.Spec.Data.Template.Metadata == nil {
+		if fleets[fleetIndex].Spec == nil || fleets[fleetIndex].Spec.Data.Template.Metadata == nil {
 			continue
 		}
 
-		devices, err := d.devStore.ListIgnoreOrg(fleet.Spec.Data.Selector.MatchLabels)
+		listParams := service.ListParams{
+			Labels: fleets[fleetIndex].Spec.Data.Selector.MatchLabels,
+			Limit:  service.MaxRecordsPerListRequest, //TODO: Handle paging
+		}
+		devices, err := d.devStore.List(ctx, orgID, listParams)
```

Comment:
I would move this work down to the database level for efficiency, allowing us to benefit from DB transactionality, etc.

1) Validate that we are only moving devices that don't belong to the fleet already or are orphaned:
SELECT device WHERE match the tags and owner!="the-new-owner-name" to identify devices already belonging to other fleets.

2) If (1) went OK

 UPDATE device SET owner="the-new-owner-name" WHERE devices match the tags AND owner=""

Reply:
i.e.

d.devStore.UpdateOwnerRef(fleets[fleetIndex].Spec.Data.Selector.MatchLabels, fleets[fleetIndex].Name)

this would be updating the OwnerField of the model.


Then, when the model is de/serialized from/to DB, because somebody is using it, we could reconstruct the owner ref field in the metadata.